### PR TITLE
[usage] Simplify Workspace record creation for tests

### PR DESCRIPTION
**Pull request by easyCZ**
## Description

Adds a helper to construct Workspace records with default values, for tests. Helpers for other objects may be added in the future, and this current one may be extended (does not handle all fields) as need arises.

## Related Issue(s)

Fixes #

## How to test

Unit tests

## Release Notes

```release-note
NONE
```

## Documentation

NONE

### CODE REVIEW

**Comment by jankeromnes** [comment]
Thanks, taking a look! 👀

Also, FYI, the build failed with:

```
pkg/controller/reconciler_test.go:22:38: cannot use "gitpodio-gitpod-gyjr82jkfnd" (untyped string constant) as db.Workspace value in argument to dbtest.NewWorkspace (typecheck)
	workspace := dbtest.NewWorkspace(t, "gitpodio-gitpod-gyjr82jkfnd")
	                                    ^
package build failed
```

**Comment by easyCZ** [comment]
> Thanks, taking a look! 👀
> 
> Also, FYI, the build failed with:
> 
> ```
> pkg/controller/reconciler_test.go:22:38: cannot use "gitpodio-gitpod-gyjr82jkfnd" (untyped string constant) as db.Workspace value in argument to dbtest.NewWorkspace (typecheck)
> 	workspace := dbtest.NewWorkspace(t, "gitpodio-gitpod-gyjr82jkfnd")
> 	                                    ^
> package build failed
> ```

Fixed. Cherry pick from a [higher level PR](https://github.com/gitpod-io/gitpod/pull/10534) gone wrong.

File: components/usage/pkg/db/dbtest/workspace.go [hunk 1]
```diff
@@ -5,20 +5,74 @@
...
+	"fmt"
 	"github.com/gitpod-io/gitpod/usage/pkg/db"
 	"github.com/google/uuid"
+	"math/rand"
 	"testing"
 )
 
-func NewWorkspace(t *testing.T, id string) db.Workspace {
+const (
+	WorkspaceContext = `{"title":"[usage] List workspaces for each workspace instance in usage period","repository":{"cloneUrl":"https://github.com/gitpod-io/gitpod.git","host":"github.com","name":"gitpod","owner":"gitpod-io","private":false},"ref":"mp/usage-list-workspaces","refType":"branch","revision":"586f22ecaeeb3b4796fd92f9ae1ca3512ca1e330","nr":10495,"base":{"repository":{"cloneUrl":"https://github.com/gitpod-io/gitpod.git","host":"github.com","name":"gitpod","owner":"gitpod-io","private":false},"ref":"mp/usage-validate-instances","refType":"branch"},"normalizedContextURL":"https://github.com/gitpod-io/gitpod/pull/10495","checkoutLocation":"gitpod"}`
```
Comment (easyCZ) [inline components/usage/pkg/db/dbtest/workspace.go:16]: Down the line, we can make this a nicer object with helpers to serialize & deserialize. For now, we don't use it in usage so left it as raw JSON

File: components/usage/pkg/db/workspace.go [hunk 2]
```diff
@@ -51,6 +51,14 @@ func (d *Workspace) TableName() string {
 	return "d_b_workspace"
 }
 
+type WorkspaceType string
+
+const (
+	WorkspaceType_Prebuild WorkspaceType = "prebuild"
+	WorkspaceType_Probe    WorkspaceType = "probe"
```
Comment (jankeromnes) [inline components/usage/pkg/db/workspace.go:58]: FYI, I believe the type `probe` is deprecated. Although, maybe there are still old occurrences in our DB, so it's probably fine to keep it here as well.

**Review by jankeromnes — APPROVED** [review]
Thanks!

Changes look good (but wow, Go looks so verbose for creating shapes with default values 🙈) and fine to merge assuming all tests still pass. 🙌

**Pull request merged**